### PR TITLE
Cleanup Old index-N Blobs in Repository Cleanup

### PR DESCRIPTION
**Pull request by original-brownbear**
Repository cleanup didn't deal with outdated index-N at the repository root, this change adds
cleaning up old index-N found in the repository.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java [hunk 1]
```diff
@@ -107,4 +111,40 @@ private String startBlockedCleanup(String repoName) throws Exception {
         waitForBlock(masterNode, repoName, TimeValue.timeValueSeconds(60));
         return masterNode;
     }
+
+    public void testCleanupOldIndexN() throws ExecutionException, InterruptedException {
+        internalCluster().startNodes(Settings.EMPTY);
+
+        final String repoName = "test-repo";
+        logger.info("-->  creating repository");
```
Comment (tlrx) [inline server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java]: I'm not sure such info logs are very helpful

**Review by tlrx — APPROVED** [review]
LGTM - left minor comment about log traces that I don't find very useful and fills the test execution logs

File: server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java (hunk 1 above)
Comment (original-brownbear) [reply server/src/test/java/org/elasticsearch/repositories/blobstore/BlobStoreRepositoryCleanupIT.java]: I gotta disagree here :) I find these super helpful when it comes to debugging test failures from uncaught exceptions (esp. failing asserts on some non-test thread) where without these you often have a very hard time figuring out where exactly you were in the test at the time of the failure.

I wonder if we could do a better job on the loggin gside though ... there's not much point actually logging this stuff to the output on passing tests (basically like we have on the CLI). We could bring that up with core/infra?

**Comment by original-brownbear** [comment]
Thanks Tanguy! Let's look into a better approach to logging elsewhere IMO, hope you don't mind :)

**Pull request merged**